File: nodejs/7 kyu/95.js
// https://www.codewars.com/kata/554e4a2f232cdd87d9000038/train/javascript

// Passed

function dnaStrand(dna) {
  const dnaMap = {
    A: "T",
    C: "G",
    T: "A",
    G: "C",
  };

  let complementDna = dna
    .split("")
    .map((symbol) => dnaMap[symbol])
    .join("");
  return complementDna;
}

const output = dnaStrand("ATTGC");
console.log(output);

File: python/7 kyu/95.py
# https://www.codewars.com/kata/56d46b8fda159582e100001b/train/python

# Passed

def travel(total_time, run_time, rest_time, speed):
    distance_covered = 0
    while total_time > 0:
        remaining_run_time = min(run_time, total_time)
        distance_covered += speed * remaining_run_time
        total_time -= run_time + rest_time

    return distance_covered

output = travel(35869784, 90, 100, 5)
print(output)